### DISCUSSION TITLE
Unable to transfer data from Client side to server side

### DISCUSSION THREAD

**Discussion by john0017** [miguelgrinberg/Flask-SocketIO#1702] [Q&A]
Hi, 
Hope you are well. 
I have a strange issue with my application. 

Some background:
> Everything works fine on localhost
> The problem is on the live version
> My test site is hosted on pythonanywhere at the moment
> i am on python3.9 on the live version and 3.8 on localhost

--Server Side--

```
  @socketio.on('connect')
  def connect():
      output = []
      print("client connected")
  
  @socketio.on('message')
  def message(msg):
      print("\n\nmessage: ", msg['data'])
```

--Client Side--

```
        let socket = io();

        socket.on('connect', () => {
            console.log('connected!!!')
            socket.emit('message', {data: 'incoming'});
        });
```

--Output--

> client side console prints "connected!!!"
> server side outputs "client connected"
> no errors

Problem:
> on 'connect' runs fine
> the message event isn't triggered and there is no output
> I have another event (not shown here) that fails to transfers data to the server as well

Any idea what is wrong? 

Thank you!

**Comment by miguelgrinberg** [comment]
Have you looked at the logs? That may have some clues. See the Troubleshooting section of the documentation if you don't know how to get logs.

**Reply by john0017** [reply]
`logs = SocketIO(logger=True, engineio_logger=True)
print(logs)`

I added the above and it returns a flask_socketio object.

**Reply by john0017** [reply]
`Tue Oct 19 13:46:49 2021 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /socket.io/?EIO=4&transport=websocket&sid=T9iVOP2V4Ftz8NbGAAAA (ip xxxxxx) !!!
2021-10-19 13:46:49 Tue Oct 19 13:46:49 2021 - uwsgi_response_write_headers_do(): Broken pipe [core/writer.c line 248] during GET /socket.io/?EIO=4&transport=websocket&sid=T9iVOP2V4Ftz8NbGAAAA (xxxxx)`

This was printed also

**Reply by miguelgrinberg** [reply]
The `logger` and `engineio_logger` options need to be added to your `SocketIO` server object, they are not added on a separate object.

**Reply by john0017** [reply]
Hi Miguel, 

Thank you very much for your help. Turns out pythonanywhere doesn't support web-sockets unfortunately. I will move on to Heroku and open a new thread if the issue persists.
